### PR TITLE
Fix redox support

### PR DESCRIPTION
**Pull request by jackpot51**

### CODE REVIEW

File: src/size.rs [hunk 1]
```diff
@@ -58,6 +61,8 @@ pub fn terminal_size() -> io::Result<(usize, usize)> {
     }
 
     Ok((try!(get_int("COLUMNS")), try!(get_int("LINES"))))
+    */
+    Ok((128,48))
```
Comment (jackpot51) [inline src/size.rs:65]: This is temporary, it maps to a 1024x768 screen

File: src/raw.rs [hunk 2]
```diff
@@ -4,14 +4,14 @@ use std::ops::{Deref, DerefMut};
 /// A terminal restorer, which keeps the previous state of the terminal, and restores it, when
 /// dropped.
 #[cfg(target_os = "redox")]
-pub struct RawTerminal<W> {
+pub struct RawTerminal<W: Write> {
```
Comment (ticki) [inline src/raw.rs:7]: This bound is unnecessary, and non-rustic.

File: src/raw.rs [hunk 3]
```diff
@@ -4,14 +4,14 @@ use std::ops::{Deref, DerefMut};
...
     output: W,
 }
 
 #[cfg(target_os = "redox")]
 impl<W: Write> Drop for RawTerminal<W> {
     fn drop(&mut self) {
-        use TermControl;
+        use control::TermWrite;
```
Comment (ticki) [inline src/raw.rs:14]: Oh, good catch.

File: src/raw.rs [hunk 4]
```diff
@@ -21,27 +21,27 @@ use termios::Termios;
 /// A terminal restorer, which keeps the previous state of the terminal, and restores it, when
 /// dropped.
 #[cfg(not(target_os = "redox"))]
-pub struct RawTerminal<W> {
+pub struct RawTerminal<W: Write> {
```
Comment (ticki) [inline src/raw.rs:24]: Again, unnecessary bound.

File: src/raw.rs [hunk 5]
```diff
@@ -21,27 +21,27 @@ use termios::Termios;
...
     prev_ios: Termios,
     output: W,
 }
 
 #[cfg(not(target_os = "redox"))]
-impl<W> Drop for RawTerminal<W> {
+impl<W: Write> Drop for RawTerminal<W> {
```
Comment (ticki) [inline src/raw.rs:30]: Unnecessary bound. You don't use any method from Write here.

File: src/raw.rs [hunk 6]
```diff
@@ -21,27 +21,27 @@ use termios::Termios;
...
     fn drop(&mut self) {
         use termios::set_terminal_attr;
         set_terminal_attr(&mut self.prev_ios as *mut _);
     }
 }
 
-impl<W> Deref for RawTerminal<W> {
+impl<W: Write> Deref for RawTerminal<W> {
```
Comment (ticki) [inline src/raw.rs:37]: Here too.

File: src/raw.rs [hunk 7]
```diff
@@ -21,27 +21,27 @@ use termios::Termios;
...
     type Target = W;
 
     fn deref(&self) -> &W {
         &self.output
     }
 }
-impl<W> DerefMut for RawTerminal<W> {
+impl<W: Write> DerefMut for RawTerminal<W> {
```
Comment (ticki) [inline src/raw.rs:44]: and here.

File: src/raw.rs [hunk 8]
```diff
@@ -58,7 +58,7 @@ impl<W: Write> Write for RawTerminal<W> {
 }
 
 /// Types which can be converted into "raw mode".
-pub trait IntoRawMode: Sized {
+pub trait IntoRawMode: Write + Sized {
```
Comment (ticki) [inline src/raw.rs:61]: This bound is fine.

**Comment by jackpot51** [comment]
The bounds on drop are required to match the struct bounds, try compiling without them

**Comment by jackpot51** [comment]
Redox compilation did not work without a Write trait bound

**Comment by ticki** [comment]
Thanks!

**Pull request merged**